ts/src/renderer/phaser/classes/PhaserAnimatedEntity.ts

```diff
@@ -11,18 +11,48 @@ class PhaserAnimatedEntity extends PhaserEntity {
 
 		const bounds = entity._bounds2d;
 		const sprite = this.sprite = scene.add.sprite(0, 0, key);
+
 		sprite.setDisplaySize(bounds.x, bounds.y);
 		sprite.rotation = entity._rotate.z;
 
 		Object.assign(this.evtListeners, {
-			'play-animation': entity.on('play-animation', this.playAnimation, this)
+			'play-animation': entity.on('play-animation', this.playAnimation, this),
+			size: entity.on('size', this.size, this),
 		});
 	}
 
 	protected playAnimation (animationId: string): void {
 		this.sprite.play(`${this.key}/${animationId}`);
 	}
 
+	protected transform (data: {
+		x: number;
+		y: number;
+		rotation: number
+	}): void {
+		this.sprite.setPosition(data.x, data.y);
+		this.sprite.rotation = data.rotation;
+	}
+
+	protected hide (): void {
+		super.hide();
+		this.sprite.setVisible(false);
```

Comment:
should this be `gameObject` instead of `sprite` as well? or both?

Reply:
probably only `gameObject`, similar to transform

Reply:
`Phaser.GameObjects.GameObject` does not have method `.setVisible()` so I gave `setActive()` to `PhaserEntity` and called `super` then `sprite.setVisible()` for `PhaserAnimatedEntity`.